### ISSUE TITLE
Codebase context inference

### ISSUE THREAD

**Issue by deepak2431** [sourcegraph/cody#113]
Currently, we infer the codebase context from the Git clone URL of the repo. This works only if the repository is directly cloned from the parent URL.

But for these cases, it doesn't work:
1. When the clone is the forked one
2. When the clone is configured with a different host

I primarily work with the forked clone and sometimes have my host configured, too, so I have encountered this issue frequently.

Suggested solution:
1. Show an info message: "Your repository cannot be configured from the Git URL". 
2. Add a link as "Add manually" inline with the repo embeddings in the chat on which the user can click and configure in the settings. An alternative can be to show a text field in the editor to add the Git repo parent URL, which can be used with the codebase.

**Comment by deepak2431** [comment]
@abeatrix @toolmantim What do you think of this?

**Comment by deepak2431** [comment]
@toolmantim Your views on this? I think you started to work on this a bit before too.

**Comment by toolmantim** [comment]
Are you needing to be able to configure a different git remote (instead of `origin`) as the codebase, for embeddings to work?

I wonder if we can read something from the local git config to start (no UI required) — `git config cody.remote "some-other-remote"`

🤔

**Comment by deepak2431** [comment]
No, I want to configure the `git remote origin`,  only but this can be different in the case of forked repo so we couldn't infer the embeddings successfully.
We can try doing what you suggested or maybe read the `upstream `remote URL.
